src/Analyzer.Utilities/JsonLineNumberResolver.cs

```diff
@@ -37,8 +39,23 @@ public JsonLineNumberResolver(TemplateContext templateContext)
         /// or 1 if it can't be determined.</returns>
         public int ResolveLineNumber(string pathInExpandedTemplate)
         {
-            JToken expandedTemplateRoot = this.templateContext.ExpandedTemplate;
-            JToken originalTemplateRoot = this.templateContext.OriginalTemplate;
+            var rootTemplateContext = this.templateContext;
+            while (rootTemplateContext != null && !rootTemplateContext.IsMainTemplate)
+            {
+                rootTemplateContext = rootTemplateContext.ParentContext;
+            }
+
+            if (rootTemplateContext == null)
+            {
+                throw new ArgumentNullException(nameof(rootTemplateContext));
+            }
+            if (!rootTemplateContext.IsMainTemplate)
+            {
+                throw new FileNotFoundException(nameof(rootTemplateContext));
```

Comment:
I don't think this line of code is reachable.  If rootTemplateContext.IsMainTemplate is false, then the while statement above wouldn't stop, and it would try to go to the next level instead, resulting in rootTemplateContext being null.